Index: lib/command/sfCommandManager.class.php
===================================================================
--- lib/command/sfCommandManager.class.php	(revision 14050)
+++ lib/command/sfCommandManager.class.php	(working copy)
@@ -328,7 +328,7 @@
   {
     if (false !== strpos($argument, '='))
     {
-      list($name, $value) = explode('=', $argument);
+      list($name, $value) = explode('=', $argument, 2);
 
       if (!$this->optionSet->hasOption($name))
       {
